### DISCUSSION TITLE
simulate a water level float

### DISCUSSION THREAD

**Discussion by Ptrvh** [arendst/Tasmota#20432] [Q&A] [answered]
Hi There,

I have a waterpump which pumps the water out my drainage pit in the sewer. I want to activate the pump when the water is at a high level and deactivate the pump on low level. I have the following idea:

I have a Sonoff Basic flashed with Tasmota.

I make a "switch" on the bottom (connect a wire to GPIO2 and GRND) and make these in the pit almost at the bottom. Next to that I make a "switch" at the top with a wire connected to the button and GRND).

I setup my Sonoff Basic as follow:
![tasmota setup](https://github.com/arendst/Tasmota/assets/22859779/04a80792-5ce0-4954-81a1-ca01b5d62480)

The relay 2 is a dummy.

Then I created the Rule: 
rule1 
on switch1#state=1 do power1 on endon 
on switch1#state=0 do power2 off endon
on switch2#state=0 do power1 off endon 
on switch2#state=1 do power2 on endon

But.....
Switch1 will control relay1 and switch2 will control relay2. 

The rules seem not to work.

Anybody an idea what is wrong or another idea to get my pump turned on and off in the way I want?

**Comment by sfromis** [comment]
Well, to follow what happens, do follow the details in the console. Then you can see what rules are firing (or not) when you activate the Switch inputs.

And make sure to have selected a SwitchMode for both to match how you expect it to work. And just to be sure, make sure that your rule set is activated too.

**Reply by Ptrvh** [reply, marked as answer]
Thanks. I do not know which switchmode they are default but switching them to Switchmode15 and then to switchmode1 did the trick. Works perfect now!

**Reply by barbudor** [reply]
default switchmode is 0 (same as 7) where the switch sends TOGGLE (2) at each change

**Reply by Ptrvh** [reply]
:)
thanks. Next issue i need to solve: seems that a water connection does not trigger the state change. probably not enough power to measure..

**Reply by sfromis** [reply]
With fresh water, the resistance is too high to work directly with a TTL input, without amplification. The common "rain" sensors use analog input instead.

**Reply by sfromis** [reply]
However, I thought that it was about simulating a float, which does not work by electrical connection through the liquid.

**Reply by Ptrvh** [reply]
The idea arised from the common rain sensors indeed. So maybe I can connect two rain sensors instead to make it work.

found this [module](https://www.aliexpress.com/item/1005003238912561.html?spm=a2g0n.shoppingtesla.product.d0.3fe4kP4FkP4FDl&pdp_npi=3%40dis%21USD%21%241.86%21%241.62%21%21%21%21%21%4021038e8317047555028485282e91f7%2112000024797728495%21btf%21%21&afTraceInfo=1005003238912561__unknown__c_ppc_item_bridge_lp__PceKSHT__1704755503092) which does what I need.. only the smart part is missing to monitor in my home automation. Need to make such a thing with smart part 😊.

**Reply by barbudor** [reply]
That module is overkill to use with an ESP. You don't need relay... Actually it does the whole job of controlling the pump.

Depending on your mounting possibilities, I would suggest:
https://www.aliexpress.com/item/1005005648584783.html
https://www.aliexpress.com/item/1005005557891213.html
https://www.aliexpress.com/item/32916697311.html

or others, search for liquid level switch

**Reply by Ptrvh** [reply]
Ordered these and will try them. Tnx
